tools/src/launcher/Main.hs

```diff
@@ -479,33 +487,18 @@ spawnNode (path, args, mbLogPath) = do
     logNotice "Starting the node"
     -- We don't explicitly close the `logHandle` here,
     -- but this will be done when we run the `CreateProcess` built
-    -- by proc later is `system'`:
+    -- by proc later in `system'`:
     -- http://hackage.haskell.org/package/process-1.6.1.0/docs/System-Process.html#v:createProcess
-    (_, logHandle) <- liftIO $ case mbLogPath of
+    cr <- liftIO $ case mbLogPath of
         Just lp -> do
-            createDirectoryIfMissing True (directory lp)
-            (lp,) <$> openFile lp AppendMode
-        Nothing -> do
-            tempdir <- fromString <$> getTemporaryDirectory
-            -- FIXME (adinapoli): `Shell` from `turtle` was giving us no-resource-leak guarantees
-            -- via the `Managed` monad, which is something we have lost here, and we are back to manual
-            -- resource control. In this case, however, shall we really want to nuke the file? It seems
-            -- something useful to have lying around in the filesystem. We should probably close the
-            -- `Handle`, though, but if this program is short lived it won't matter anyway.
-            IO.openTempFile tempdir "cardano-node-output.log"
-    -- TODO (jmitchell): Find a safe, reliable way to print `logPath`. Cardano
-    -- fails when it prints unicode characters. In the meantime, don't print it.
-    -- See DAEF-12.
-
-    -- printf ("Redirecting node's stdout and stderr to "%fp%"\n") logPath
-    liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
-    let cr = (Process.proc (toString path) (map toString args))
-                 { Process.std_in  = Process.CreatePipe
-                 , Process.std_out = Process.UseHandle logHandle
-                 , Process.std_err = Process.UseHandle logHandle
-                 }
+            createLogFileProc path args lp
+            -- TODO (jmitchell): Find a safe, reliable way to print `logPath`. Cardano
+            -- fails when it prints unicode characters. In the meantime, don't print it.
+            -- See DAEF-12.
+            -- printf ("Redirecting node's stdout and stderr to "%fp%"\n") logPath
+        Nothing -> return $ createProc Process.CreatePipe path args
```

Comment:
Once again, see this comment: https://github.com/input-output-hk/cardano-sl/pull/1629#discussion_r155955761

It should be `CreatePipe` when wallet logging is enabled (and not redirected) and `Inherit` otherwise. So you need to pass one more argument to this function.
The same is true for wallet.